src/main/java/eu/dissco/core/translator/component/MessageCompressionComponent.java

```diff
@@ -0,0 +1,78 @@
+package eu.dissco.core.translator.component;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.stream.Collectors;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.amqp.core.Message;
+import org.springframework.amqp.core.MessageProperties;
+import org.springframework.amqp.support.converter.MessageConversionException;
+import org.springframework.amqp.support.converter.MessageConverter;
+import org.springframework.amqp.support.converter.SimpleMessageConverter;
+import org.springframework.stereotype.Component;
+
+@Component
+@Slf4j
+public class MessageCompressionComponent implements MessageConverter {
+
+  final MessageConverter simpleConverter = new SimpleMessageConverter();
+
+  private static byte[] deflateMessage(byte[] message) throws IOException {
+    final int BUFFER_SIZE = 8192; // 8KB
+    try (var rstBao = new ByteArrayOutputStream(BUFFER_SIZE)) {
```

Comment:
maybe define these two resources on one line? 
Also, can you make `rstBao` a more descriptive variable name? what's rst?

Reply:
Ok yeah, so there is a curious thing with the two resources. The first needs to be closed so that it pushes out the last byts before you can close the outer outputstream. If you combine them into one statement it will close both at the same time and the last bytes are lost, creating corrupt messages :+1:  Took me a moment to figure out. I still wanted to auto close these things so this solved it. Might add a comment to clarify this